ide/nativeimage.api/src/org/netbeans/modules/nativeimage/api/debug/NIDebugger.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.nativeimage.api.debug;
+
+import java.io.File;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+
+import org.netbeans.api.extexecution.ExecutionDescriptor;
+import org.netbeans.modules.nativeimage.spi.debug.NIDebuggerProvider;
+import org.netbeans.modules.nativeimage.spi.debug.filters.FrameDisplayer;
+import org.netbeans.modules.nativeimage.spi.debug.filters.VariablesDisplayer;
+
+/**
+ * Representation of a native image debugger.
+ */
+public final class NIDebugger {
+
+    private final NIDebuggerProvider provider;
+
+    NIDebugger(NIDebuggerProvider provider) {
+        this.provider = provider;
+    }
+
+    /**
+     * Add or change a line breakpoint into the debugger.
+     * A breakpoint is added when the `id` is used for the first time and modified
+     * when breakpoint with that `id` was added already.
+     *
+     * @param id a unique ID of the breakpoint
+     * @param filePath a path of the file to break in
+     * @param line a 1-based line to to break at
+     * @param enabled whether the breakpoint is initially enabled
+     * @param condition a possible breakpoint condition
+     * @param hidden <code>true</code> when the resulting native breakpoint
+     *               should be visible in the UI, <code>false</code> otherwise
+     * @return an instance of the native breakpoint
+     */
+    public Object addLineBreakpoint(Object id, String filePath, int line, boolean enabled, String condition, boolean hidden) {
+        return provider.addLineBreakpoint(id, filePath, line, enabled, condition, hidden);
+    }
+
+    /**
+     * Remove breakpoint with the given id.
+     *
+     * @param id the ID of the breakpoint to remove
+     */
+    public void removeBreakpoint(Object id) {
+        provider.removeBreakpoint(id);
+    }
+
+    /**
+     * Displayer of native frames.
+     *
+     * @param frameDisplayer translator of the native frame to it's displayed information
+     */
+    public void setFrameDisplayer(FrameDisplayer frameDisplayer) {
```

Comment:
`FrameDisplayer` and `VariablesDisplayer` (btw. one is _singular_ and second is in _plural_ form!?) are one time setters - it'd be better to provide them when building the debugger. What about:
```
NIDebugger.newBuilder().
  frameDisplayer(...).
  variableDisplayer(...).
  build();
```
The builder pattern turned out to work quite a well for one time, optional properties.

Reply:
Done.